### DISCUSSION TITLE
getStaticPaths, getStaticProps, ssrCache

### DISCUSSION THREAD

**Discussion by mohux** [vercel/next.js#11481] [Help]
Error [ERR_HTTP_HEADERS_SENT]: Cannot set headers after they are sent to the client

this is what  am getting when am using ssrCaching example

https://github.com/zeit/next.js/tree/canary/examples/ssr-caching

I had to change one line in server.js

from 
```
const pagePath = '/blog';
```
to
```
const pagePath = `/blog/${req.params.id}`;
```
in order to make refresh work

but it resulted in receiving the mentioned error

if I swap to getInitialProps
it works without any error

does it have anything to do with caching built inside SSG?

**Comment by transitive-bullshit** [comment]
This looks like a duplicate of https://github.com/vercel/next.js/issues/14186 that has been fixed in `next@canary` as of about a week ago.
